### PR TITLE
Fix infinite loop in W3DDisplay::Take_ScreenShot

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/game/common/globaldata.h [hunk 1]
```diff
@@ -63,6 +63,8 @@ class GlobalData : public SubsystemInterface
 
     bool Set_Time_Of_Day(TimeOfDayType time);
 
+    const char *Get_Path_User_Data() const { return m_userDataDirectory.Str(); }
```
Comment (xezon) [inline src/game/common/globaldata.h]: Seen in World Builder

File: src/platform/w3dengine/client/w3ddisplay.cpp [hunk 2]
```diff
@@ -1434,19 +1434,18 @@ void Create_Bmp_File(const char *filename, const char *data, int width, int heig
...
+    while (true) {
         sprintf(buf, "%s%.3d.bmp", "sshot", currentShot);
         currentShot++;
-        strcpy(fname, g_theWriteableGlobalData->m_userDataDirectory);
+        strcpy(fname, g_theWriteableGlobalData->Get_Path_User_Data());
         strcat(fname, buf);
 
-        if (_access(fname, 0) != -1) {
-            found = true;
+        if (_access(fname, 0) == -1) {
```
Comment (duncanspumpkin) [inline src/platform/w3dengine/client/w3ddisplay.cpp:1447]: you could just make this a do while and skip the while(true).

Comment (xezon) [reply src/platform/w3dengine/client/w3ddisplay.cpp:1447]: I think it's fine as is. Compiler will know what to do.

File: src/game/common/globaldata.h (hunk 1 above)
Comment (tomsons26) [reply src/game/common/globaldata.h]: Should actually return AsciiString instance

File: src/platform/w3dengine/client/w3ddisplay.cpp (hunk 2 above)
Comment (tomsons26) [reply src/platform/w3dengine/client/w3ddisplay.cpp:1447]: found shouldn't be removed, however "!= -1" should be "== -1" yeah

File: src/game/common/globaldata.h (hunk 1 above)
Comment (xezon) [reply src/game/common/globaldata.h]: Very well.

File: src/game/common/globaldata.h [hunk 3]
```diff
@@ -63,6 +63,8 @@ class GlobalData : public SubsystemInterface
 
     bool Set_Time_Of_Day(TimeOfDayType time);
 
+    Utf8String Get_Path_User_Data() const { return m_userDataDirectory.Str(); }
```
Comment (jonwil) [inline src/game/common/globaldata.h]: This seems wrong, it shouldn't be calling .Str() if its returning Utf8String.
Fix this and it should be good.

**Review by jonwil — CHANGES_REQUESTED** [review]

File: src/game/common/globaldata.h (hunk 3 above)
Comment (xezon) [reply src/game/common/globaldata.h]: Fixed.

**Review by jonwil — APPROVED** [review]

**Pull request merged**